updated-sdk-example (#392)
Consensys/doc.zk-evm: docs/build-on-linea/linea-sdk/index.md

 5. Claiming messages (use one of the get message methods to grab all the parameters values)

 
:::info[New Features]

>  The updated Linea SDK package enhances the L1 contract interaction and adds support for the new L1 claiming system, which is based on a merkle tree.

Reviewer: based on a `Merkle` tree `and requires a Merkle proof for claiming`

Author: fixed

>  There are two important things to note:

Reviewer: An additional point can be made here as well (so there are "three important things to note")

The Previous L1 claiming function and code samples provided here caters for the transition period where pre-transition messages are claimed without the Merkle proof and post-transition with proof. If this SDK is being consumed after the transition, using the logic that switches between Merkle and non-Merkle proof claiming is sub-optimal.

Author: added

>   - The previous L1 claiming and all function associated are still supported.

Reviewer: `functions` vs. `function`

  - The L2 claiming remains unaltered, and all SDK features to interact with L2 will remain unchanged.

 The updated SDK introduces several new features for L1 interactions:

 A new L1ClaimingService class that includes the following functions:
  - getMessageProof: This function retrieves the message merkle tree proof required for new message claims on L1.
  - isClaimingNeedingProof: This function determines whether a proof is needed to claim a message.
  - getMessageStatus: This function retrieves a message's status, returning the status of both old and new messages.
  - estimateClaimMessageGas: This function provides an estimate of the gas cost for both old and new claim transactions.
  - claimMessage: This function enables a message to be claimed using either the old or new method.

 Two new functions in the L1 contract:

>   - estimateClaimWithProofGas to estimates the gas cost for new claim transactions.

Reviewer: `estimate` vs. `estimates`

Reviewer: The formatting is generally inconsistent.

e.g. ``- getMessageProof:` has a colon, some places don't like in this one.

Some lines have periods. Some don't.

Some places we call it `functions` and some places `methods` - probably best to align on one.

Author: fixed

 5. Claiming messages (use one of the get message methods to grab all the parameters values)

 
:::info[New Features]

>  The updated Linea SDK package enhances the L1 contract interaction and adds support for the new L1 claiming system, which is based on a Merkle tree and requires a Merkle proof for claiming.

Reviewer 2: ```suggestion
The updated Linea SDK package enhances the L1 contract interaction and adds support for the new L1 claiming system, which is based on a Merkle tree and requires a Merkle proof for claiming.
```

 
 There are three important things to note:

>   - The previous L1 claiming and all functions associated are still supported.

Reviewer 2: If it's still supported, wouldn't it be better described as "existing" rather than "previous"? 

Also, I think we're missing a noun after "claiming": claiming process, claiming flow, claiming mechanism, etc.

>   - The L2 claiming remains unaltered, and all SDK features to interact with L2 will remain unchanged.

Reviewer 2: ```suggestion
  - The L2 claiming remains unaltered, and all SDK features for interacting with L2 will remain unchanged.
```

>   - The previous L1 claiming function and code samples provided here cater to the transition period where pre-transition messages are claimed without the Merkle proof and post-transition with proof. If this SDK is being used after the transition, using the logic that switches between Merkle and non-Merkle proof claiming is sub-optimal.

Reviewer 2: I think this whole bullet needs some work for clarity. It's quite hard to follow. I don't have any concrete suggestions as I'm not entirely sure what it's trying to say, but I think the use of "pre-transition" and "post-transition" is potentially adding complexity. Let me know if you want me to take a second look and devote more time to this.

>  The updated SDK introduces several new features for L1 interactions:

Reviewer 2: ```suggestion
The updated SDK introduces several new features for L1 interactions:
```

>  A new L1ClaimingService class that includes the following functions:

Reviewer 2: ```suggestion
A new L1ClaimingService class that includes the following functions:
```

  - getMessageProof: This function retrieves the message Merkle tree proof required for new message claims on L1.
  - isClaimingNeedingProof: This function determines whether a proof is needed to claim a message.
  - getMessageStatus: This function retrieves a message's status, returning the status of both old and new messages.
  - estimateClaimMessageGas: This function provides an estimate of the gas cost for both old and new claim transactions.
  - claimMessage: This function enables a message to be claimed using either the old or new function.

>  Two new functions in the L1 contract:

Reviewer 2: ```suggestion
Two new functions in the L1 contract:
```

  - The previous L1 claiming function and code samples provided here cater to the transition period where pre-transition messages are claimed without the Merkle proof and post-transition with proof. If this SDK is being used after the transition, using the logic that switches between Merkle and non-Merkle proof claiming is sub-optimal.

 The updated SDK introduces several new features for L1 interactions:

 A new L1ClaimingService class that includes the following functions:
  - getMessageProof: This function retrieves the message Merkle tree proof required for new message claims on L1.
  - isClaimingNeedingProof: This function determines whether a proof is needed to claim a message.
  - getMessageStatus: This function retrieves a message's status, returning the status of both old and new messages.
  - estimateClaimMessageGas: This function provides an estimate of the gas cost for both old and new claim transactions.
  - claimMessage: This function enables a message to be claimed using either the old or new function.

 Two new functions in the L1 contract:
  - estimateClaimWithProofGas: This functions estimates the gas cost for new claim transactions.

>   - claimWithProof: This functions claims a message using the new claimMessageWithProof function.

Reviewer: this function vs this functions
